typespec-extension/src/code-model-builder.ts

```diff
@@ -1021,17 +1007,20 @@ export class CodeModelBuilder {
     } else {
       // schema
       let schema;
+      const sdkType = getClientType(this.sdkContext, param.param);
       if (
         param.type === "header" &&
         param.param.type.kind === "Scalar" &&
         getEncode(this.program, param.param) === undefined &&
         getEncode(this.program, param.param.type) === undefined &&
-        (hasScalarAsBase(param.param.type, "utcDateTime") || hasScalarAsBase(param.param.type, "offsetDateTime"))
+        (hasScalarAsBase(param.param.type, "utcDateTime") || hasScalarAsBase(param.param.type, "offsetDateTime")) &&
```

Comment:
nit, can this be `sdkType.type`?

Reply:
Yes, it should be removed. This part is related with operation, so my original plan is to remove those TSP related types in processing operation after switch to `SdkPackage` (getAllOperation)